packages/smpc/tensor/share_tensor.py

```diff
@@ -1,9 +1,10 @@
 from syft.core.tensor.passthrough import PassthroughTensor
+from .share_tensor_ancestor import ShareTensorAncestor
 
 # TODO: This is not secure
 import random
 
-class ShareTensor(PassthroughTensor):
+class ShareTensor(PassthroughTensor, ShareTensorAncestor):
```

Comment:
ShareTensor won't ever have ShareTensor in its chain i don't think so no need to subclass it from ancestor